packages/modeling/src/operations/extrusions/extrudeLinearGeom2.js

```diff
@@ -32,16 +32,13 @@ export const extrudeGeom2 = (options, geometry) => {
     twistSteps = 1
   }
 
-  // convert to vector in order to perform transforms
-  const offsetv = vec3.clone(offset)
-
   let baseSlice = slice.fromGeom2(geometry)
-  if (offsetv[2] < 0) baseSlice = slice.reverse(baseSlice)
+  if (offset[2] < 0) baseSlice = slice.reverse(baseSlice)
```

Comment:
This looks fine but i think this whole function needs some changes to improve performance. This function is not exposed so all parameters should be checked by calling functions, and passed to extrudeGeom2 with the correct type. Also extrudeGeom2 should be renamed to match the file name. etc. etc.

Reply:
I'm happy to make that change in another PR

Reply:
Cool. How about a new issue to track?